cmd/destroy/destroy.go

```diff
@@ -263,11 +263,18 @@ func (dc *destroyCommand) runDestroy(ctx context.Context, opts *Options) error {
 	os.Setenv(constants.OktetoNameEnvVar, opts.Name)
 
 	if opts.DestroyDependencies {
-		for depName := range manifest.Dependencies {
+		for depName, depInfo := range manifest.Dependencies {
 			oktetoLog.SetStage(fmt.Sprintf("Destroying dependency '%s'", depName))
+
+			namespace := okteto.Context().Namespace
+			if depInfo.Namespace != "" {
+				namespace = depInfo.Namespace
+			}
```

Comment:
Why are we checking this for the destroy but not for the deploy? Is there any difference? I think it is not needed with the refactor and the logic in setDefaults, right?

Reply:
Tested and it was also needed in the deploy command. I've added it also to the deploy command